core/baseline-core-v1.0-psd01.md

```diff
@@ -637,36 +637,418 @@ Describes the mechanisms ensuring that other parties (i.e parties outside of tra
 
-| Requirement ID|Requirement  |
-| :--- | :--- |
-| AGEXEC1 | IBaselineRPC  |
-| AGEXEC2| Shared agreement: up to each workgroup to agree on business logic  |
-| AGEXEC3| Shielding: up up to each workgroup to determine a suitable shielding mechanism. |
+<figure>
+  <img
+  src="./images/Baseline-Fig-6.1-Workflow-Agreement-Execution.png"
+  >
+  <figcaption>Figure 6.1: Conceptual View of Workflow, Workstep and Workgroup</figcaption>
+</figure>
 
-## 6.2 Business Logic Execution
+## 6.1 BPI Workstep
 
-Describes the mechanisms supporting the execution of the agreed business logic.
+First, we will discuss the requirements for worksteps which will be implemented in the Virtual State Machine of the BPI Processing Layer. Note that strictly speaking we need to differentiate between the workstep as a logical construct, and its instantiation within a BPI which we call a workstep instance. In the following, and unless required for disambiguation, we shall use workstep also to mean workstep instance. 
 
+**[RXX]** A workstep MUST have an input, one or more process steps and an output.
 
-| Requirement ID|Requirement  |
-| :--- | :--- |
-| AGEXEC4 | Workflows, worksteps - business rules chaining and sequenced execution |
-| AGEXEC5| Deterministic result - given a set of arguments and a state of the ledger, execution of business logic must produce the same result. |
-| AGEXEC6| Event-driven |
-| AGEXEC7| On-chain/ DTL execution mode|
+This is just a well known convention from business process management frameworks.
 
-## 6.3 Performance
+**[RXX]** The input of a workstep MUST represent a new, proposed state of a state object compliant with the agreement between the agreement counterparties.
 
-Describes the performance requirements for the Agreement Execution component.
+**[RXX]** The process steps in a workstep MUST represent a verification system comprised of the set, or subset, of agreement rules and agreement data such that an input can be validated to comply with the agreement rules and agreement data, or not.
 
-| Requirement ID|Requirement  |
-| :--- | :--- |
-| AGEXEC8 | Processing/Finality time  |
-| AGEXEC9| Execution/Process monitoring |
+**[RXX]** The output of a workstep MUST represent the verifiable validation result of an input into a workstep as a correct new agreement state.
+
+Note that a new agreement state after a correct workstep execution is defined as 
+```
+New Agreement State = Old Agreement State + Agreed upon New State Object + Workstep Output 
+```
+
+See sections Xx and YY below on BPI accounts associated with state objects and BPI transactions for details.
```

Comment:
Xx Yy :)

Reply:
Merging this but certainly we will have another round of PRs to address outstanding cleanup items.